### PR TITLE
Create HYDRA provider

### PR DESCRIPTION
**Pull request by jonapap**
Will merge as is for now. Other PRs will do the following:
- [ ] Add Docker support
- [ ] Send to a RabbitMQ instance
- [ ] Log all messages to a local file
- [ ] Store the offset between the board's epoch and the Unix epoch
- [ ] Receive logs from the boards (optional, may go in a separate PR)
- [ ] Add CI

Using async since the most popular RabbitMQ library is also async (https://crates.io/crates/lapin). I was hesitant to do this since this is not a super complicated app, but I think using async will be helpful.

### CODE REVIEW

File: hydra_provider/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+        available_ports()
+            .context("No serial port specified and couldn't retrieve available ports")?
+            .iter()
+            .filter(|x| x.port_name.contains("USB"))
+            .last()
+            .context("No serial port specified and couldn't find any port")?
+            .port_name
+            .clone()
+    };
```
Comment (ajayyy) [inline hydra_provider/src/main.rs:41]: Thanks gpt3 :)

image

image

**Review by ajayyy — APPROVED** [review]

File: hydra_provider/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            Err(err) => info!("Error reading message: {err}"),
+        }
+    }
+}
+
+async fn read_message<T: AsyncBufRead + Unpin>(reader: &mut T) -> Result<Message> {
+    let mut data = vec![];
+    reader.read_until(0x0, &mut data).await?;
+
+    let msg: Message = from_bytes_cobs(data.as_mut_slice())?;
```
Comment (ajayyy) [inline hydra_provider/src/main.rs:68]: Why cobs instead of mavlink?

Comment (jonapap) [reply hydra_provider/src/main.rs:68]: Just to get a proof of concept and something to get started with. We should switch to MAVLink relatively soon.

**Comment by jonapap** [comment]
Blocked on uorocketry/hydra#20, then we can merge

**Pull request merged**